web/src/lib/components/shared-components/gallery-viewer/gallery-viewer.svelte

```diff
@@ -17,21 +19,51 @@
 	let currentViewAssetIndex = 0;
 
 	let viewWidth: number;
-	let thumbnailSize = 300;
+	let justifiedLayoutResult: any;
+	const geoArray: Array<number> = [];
 
-	$: isMultiSelectionMode = selectedAssets.size > 0;
+	onMount(() => {
+		getAssetRatio();
+		buildJustifiedLayout();
+	});
 
 	$: {
-		if (assets.length < 6) {
-			thumbnailSize = Math.min(320, Math.floor(viewWidth / assets.length - assets.length));
-		} else {
-			if (viewWidth > 600) thumbnailSize = Math.floor(viewWidth / 6 - 6);
-			else if (viewWidth > 400) thumbnailSize = Math.floor(viewWidth / 4 - 6);
-			else if (viewWidth > 300) thumbnailSize = Math.floor(viewWidth / 2 - 6);
-			else if (viewWidth > 200) thumbnailSize = Math.floor(viewWidth / 2 - 6);
-			else if (viewWidth > 100) thumbnailSize = Math.floor(viewWidth / 1 - 6);
+		if (viewWidth) {
+			buildJustifiedLayout();
 		}
 	}
+	$: isMultiSelectionMode = selectedAssets.size > 0;
+
+	const getAssetRatio = () => {
+		assets.forEach((a) => {
+			if (a.exifInfo?.exifImageHeight && a.exifInfo?.exifImageWidth) {
+				const height = a.exifInfo?.exifImageHeight;
+				const width = a.exifInfo?.exifImageWidth;
+				const orientation = Number(a.exifInfo?.orientation);
+
+				if (orientation !== null && orientation !== undefined) {
+					if (orientation == 6 || orientation == -90) {
+						geoArray.push(height / width);
+						return;
+					} else {
+						geoArray.push(width / height);
+						return;
+					}
+				}
+
+				geoArray.push(width / height);
+			} else {
+				geoArray.push(1);
+			}
```

Comment:
I would try to avoid this else by returning 1 above and the else is  just a fall through.